File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
                private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Check", typeof(bool));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, false, DateTime.Now.AddDays(i) });
            return tbl;
        }


        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
            new ColumnInplaceEditorHelper(gridColumn1, new RepositoryItemSpinEdit());
            gridColumn1.OptionsColumn.AllowSort = DevExpress.Utils.DefaultBoolean.False;
            gridColumn1.OptionsFilter.AllowFilter = false;
            new ColumnInplaceEditorHelper(gridColumn2, new RepositoryItemSpinEdit());
            RepositoryItemCheckEdit riCheckEdit = new RepositoryItemCheckEdit();
            riCheckEdit.AllowFocused = false;
            riCheckEdit.EditValueChanged += new EventHandler(riCheckEdit_EditValueChanged);
            new ColumnInplaceEditorHelper(gridColumn3, riCheckEdit);
            new ColumnInplaceEditorHelper(gridColumn4, new RepositoryItemProgressBar()).EditValue = 50;
        }

        void riCheckEdit_EditValueChanged(object sender, EventArgs e)
        {
            try
            {
                gridView1.BeginDataUpdate();
                for (int i = 0; i < gridView1.DataRowCount; i++)
                {
                    gridView1.SetRowCellValue(i, "Check", (sender as CheckEdit).Checked);
                }
            }
            catch (Exception ex)
            {
                
            }
            finally
            {
                gridView1.EndDataUpdate();
            }
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors

Namespace WindowsApplication1

    Public Partial Class Form1
        Inherits Form

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Check", GetType(Boolean))
            tbl.Columns.Add("Date", GetType(Date))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i, False, Date.Now.AddDays(i)})
            Next

            Return tbl
        End Function

        Public Sub New()
            InitializeComponent()
            gridControl1.DataSource = CreateTable(20)
            Dim tmp_ColumnInplaceEditorHelper = New ColumnInplaceEditorHelper(gridColumn1, New RepositoryItemSpinEdit())
            gridColumn1.OptionsColumn.AllowSort = DevExpress.Utils.DefaultBoolean.False
            gridColumn1.OptionsFilter.AllowFilter = False
            Dim tmp_ColumnInplaceEditorHelper = New ColumnInplaceEditorHelper(gridColumn2, New RepositoryItemSpinEdit())
            Dim riCheckEdit As RepositoryItemCheckEdit = New RepositoryItemCheckEdit()
            riCheckEdit.AllowFocused = False
            AddHandler riCheckEdit.EditValueChanged, New EventHandler(AddressOf riCheckEdit_EditValueChanged)
            Dim tmp_ColumnInplaceEditorHelper = New ColumnInplaceEditorHelper(gridColumn3, riCheckEdit)
            New ColumnInplaceEditorHelper(gridColumn4, New RepositoryItemProgressBar()).EditValue = 50
        End Sub

        Private Sub riCheckEdit_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            Try
                gridView1.BeginDataUpdate()
                For i As Integer = 0 To gridView1.DataRowCount - 1
                    gridView1.SetRowCellValue(i, "Check", TryCast(sender, CheckEdit).Checked)
                Next
            Catch ex As Exception
            Finally
                gridView1.EndDataUpdate()
            End Try
        End Sub
    End Class
End Namespace
